test/test_strerror.cpp

```diff
@@ -54,3 +61,38 @@ TEST(test_strerror, get_error) {
     "' did not cause the expected error message.";
 #endif
 }
+
+/*
+   Define the blueprint of a mock identified by `strerror_r_proto`
+   strerror_r signature:
+   int strerror_r(int errnum, char *buf, size_t buflen);
+*/
+mmk_mock_define(strerror_r_mock, int, char *, size_t);
+
+/* Mocking test example */
+TEST(test_strerror, test_mock) {
+  /* Mock the strerror_r function in the current module using
+     the `strerror_r_mock` blueprint. */
+  // mmk_mock("strerror_r@rcutils", strerror_r_mock);
+  // mmk_mock("__xpg_strerror_r@self", strerror_r_mock);
+  // mmk_mock("strerror_r@self", strerror_r_mock);
+  mmk_mock("__xpg_strerror_r@lib:rcutils", strerror_r_mock);
```

Comment:
@Blast545 hmm the symbol is non-portable. Maybe something like:
```suggestion
  mmk_mock(RCUTILS_STRINGIFY(strerror_r) "@lib:rcutils", strerror_r_mock);
```
can do.

Reply:
You mean that we could define another macro similar to RCUTILS_STRINGIFY? I don't understand this suggestion

Reply:
I mean to let the compiler find out what `strerror_r` actual symbol is. If `strerror_r` is a macro that expands to something else, that'll get resolved before `RCUTILS_STRINGIFY` turns it into a string literal. Then we rely on C to concatenate string literals (in C, "foo" " bar" is the same as writing "foobar") and _voilà_! We get to write it in a portable way.

Reply:
Mmm, any ideas on how to do that? that would be great, not sure if the compiler is able to find that actual symbol name.

Without anything else, it resolves to "strerror_r@lib:rcutils". We could write a macro that returns the hardcoded generated name in each platform if we find how that name is generated.

Reply:
Nevermind what it expands too, I think it's more portable to let the compiler figure out what the symbol is instead of hardcoding it.